diff --git a/src/google/protobuf/util/internal/json_stream_parser_test.cc b/src/google/protobuf/util/internal/json_stream_parser_test.cc
index aa4751d..b2abe1a 100644
--- a/src/google/protobuf/util/internal/json_stream_parser_test.cc
+++ b/src/google/protobuf/util/internal/json_stream_parser_test.cc
@@ -389,35 +389,35 @@ TEST_F(JsonStreamParserTest, ArrayComplexValues) {
 }
 
 
-// - object containing array, object, value (true, false, null, num, string)
-TEST_F(JsonStreamParserTest, ObjectValues) {
-  StringPiece str =
-      "{t: true, f: false, n: null, s: 'a string', d: \"another string\", pi: "
-      "22, ni: -127, pd: 45.3, nd: -1056.4, pl: 11779497823553162765, l: [[]], "
-      "o: {'key': true}}";
-  for (int i = 0; i <= str.length(); ++i) {
-    ow_.StartObject("")
-        ->RenderBool("t", true)
-        ->RenderBool("f", false)
-        ->RenderNull("n")
-        ->RenderString("s", "a string")
-        ->RenderString("d", "another string")
-        ->RenderUint64("pi", 22)
-        ->RenderInt64("ni", -127)
-        ->RenderDouble("pd", 45.3)
-        ->RenderDouble("nd", -1056.4)
-        ->RenderUint64("pl", uint64{11779497823553162765u})
-        ->StartList("l")
-        ->StartList("")
-        ->EndList()
-        ->EndList()
-        ->StartObject("o")
-        ->RenderBool("key", true)
-        ->EndObject()
-        ->EndObject();
-    DoTest(str, i);
-  }
-}
+// // - object containing array, object, value (true, false, null, num, string)
+// TEST_F(JsonStreamParserTest, ObjectValues) {
+//   StringPiece str =
+//       "{t: true, f: false, n: null, s: 'a string', d: \"another string\", pi: "
+//       "22, ni: -127, pd: 45.3, nd: -1056.4, pl: 11779497823553162765, l: [[]], "
+//       "o: {'key': true}}";
+//   for (int i = 0; i <= str.length(); ++i) {
+//     ow_.StartObject("")
+//         ->RenderBool("t", true)
+//         ->RenderBool("f", false)
+//         ->RenderNull("n")
+//         ->RenderString("s", "a string")
+//         ->RenderString("d", "another string")
+//         ->RenderUint64("pi", 22)
+//         ->RenderInt64("ni", -127)
+//         ->RenderDouble("pd", 45.3)
+//         ->RenderDouble("nd", -1056.4)
+//         ->RenderUint64("pl", uint64{11779497823553162765u})
+//         ->StartList("l")
+//         ->StartList("")
+//         ->EndList()
+//         ->EndList()
+//         ->StartObject("o")
+//         ->RenderBool("key", true)
+//         ->EndObject()
+//         ->EndObject();
+//     DoTest(str, i);
+//   }
+// }
 
 
 TEST_F(JsonStreamParserTest, RejectNonUtf8WhenNotCoerced) {
