From 3263ba82ca089774bf0887ffeaf1e6da8d34a314 Mon Sep 17 00:00:00 2001
From: "Dougal J. Sutherland" <dougal@gmail.com>
Date: Sun, 4 Mar 2018 15:08:53 +0000
Subject: [PATCH] conda-forge: add _static suffix on windows

---
 cmake/CMakeLists.txt | 5 +++++
 cmake/protoc.cmake   | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index f159f60e..242901ac 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -130,6 +130,11 @@ else (protobuf_BUILD_SHARED_LIBS)
       endif(${flag_var} MATCHES "/MD")
     endforeach(flag_var)
   endif (MSVC AND protobuf_MSVC_STATIC_RUNTIME)
+  # conda-forge: suffix built names with _static to distinguish from shared
+  #              libs that we've installed side-by-side
+  if (MSVC)
+    set(CMAKE_RELEASE_POSTFIX "_static")
+  endif (MSVC)
 endif (protobuf_BUILD_SHARED_LIBS)
 
 if (MSVC)
diff --git a/cmake/protoc.cmake b/cmake/protoc.cmake
index 4f07c389..6e60a549 100644
--- a/cmake/protoc.cmake
+++ b/cmake/protoc.cmake
@@ -4,3 +4,8 @@ set(protoc_files
 
 add_executable(protoc ${protoc_files})
 target_link_libraries(protoc libprotobuf libprotoc)
+# conda-forge: add _static to output name on windows
+if (MSVC AND NOT protobuf_BUILD_SHARED_LIBS)
+  set_target_properties(protoc
+    OUTPUT_NAME protoc_static)
+endif(MSVC AND NOT protobuf_BUILD_SHARED_LIBS)
-- 
2.15.0

