From 812bdf8605089a0bd3a12c9ab99437bc5662177f Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 4 Sep 2022 10:57:08 +0200
Subject: [PATCH 3/3] set import lib extension on windows

---
 cmake/CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index ac92442a1..e9d0d15ff 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -307,6 +307,12 @@ if (protobuf_BUILD_PROTOC_BINARIES)
   endif (NOT DEFINED protobuf_PROTOC_EXE)
 endif (protobuf_BUILD_PROTOC_BINARIES)
 
+if (MSVC)
+  set_target_properties(libprotoc PROPERTIES IMPORT_SUFFIX "-import.lib")
+  set_target_properties(libprotobuf PROPERTIES IMPORT_SUFFIX "-import.lib")
+  set_target_properties(libprotobuf-lite PROPERTIES IMPORT_SUFFIX "-import.lib")
+endif (MSVC)
+
 # Ensure we have a protoc executable if we need one
 if (protobuf_BUILD_TESTS OR protobuf_BUILD_CONFORMANCE OR protobuf_BUILD_EXAMPLES)
   if (NOT DEFINED protobuf_PROTOC_EXE)
-- 
2.37.0.windows.1

