From d36a74a66339a77f2da3c3cd6c5f5a7c76d9194a Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 9 Aug 2023 14:06:35 +1100
Subject: [PATCH 4/7] do not install vendored gmock

---
 cmake/install.cmake | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/cmake/install.cmake b/cmake/install.cmake
index bf5650984..2dd8c5c38 100644
--- a/cmake/install.cmake
+++ b/cmake/install.cmake
@@ -169,10 +169,3 @@ if(protobuf_INSTALL_EXAMPLES)
     DESTINATION "${CMAKE_INSTALL_EXAMPLEDIR}"
     COMPONENT protobuf-examples)
 endif()
-
-if (protobuf_INSTALL_TESTS)
-  install(TARGETS gmock EXPORT protobuf-targets
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-endif()
